internal/ignition/ignition.go

```diff
@@ -1360,3 +1384,40 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func FormatRegistriesConfForIgnition(mirrorRegistriesCaConfig *models.MirrorRegistriesCaConfig) (string, string, error) {
+	if mirrorRegistriesCaConfig == nil {
+		return "", "", nil
+	}
+	caConfig := mirrorRegistriesCaConfig.CaConfig
+	mirrorRegistriesConfig := mirrorRegistriesCaConfig.MirrorRegistriesConfig
+
+	// create a map that represents TOML tree
+	treeMap := make(map[string]interface{})
+	registryEntryList := []map[string]interface{}{}
+	// loop over mirror registries data and for each create its own map
+	for _, mirrorRegistryConfig := range mirrorRegistriesConfig.MirrorRegistries {
```

Comment:
Maybe add a comment here that you're constructing the `/etc/containers/registries.conf` file and that its format is described in `man containers-registries.conf`.